src/Chirp.CSVDBService/Program.cs

```diff
@@ -16,23 +32,31 @@
 List<Cheep> Read(int? limit = null) //Reads the cheeps, whit a limit as an option
 {
     string path = "./data/chirp_cli_db.csv";
+
     if (limit <= 0)
     {
-        throw new ArgumentOutOfRangeException($"{nameof(limit)} must be greater than 0");
+        throw new ArgumentOutOfRangeException($"--limit must be greater than 0");
     }
     if (Path.Exists(path))
     {
         using (StreamReader reader = new StreamReader(path))
         using (CsvReader csv = new CsvReader(reader, CultureInfo.InvariantCulture))
         {
             List<Cheep> all_records = csv.GetRecords<Cheep>().ToList();
-            return all_records;
+            List<Cheep> records = new List<Cheep>();
+            if(limit < all_records.Count){
+                for (int i = all_records.Count-1; i >= all_records.Count- limit; i--){ //this forloop insures getting the lates cheeps
+                    records.Add(all_records[i]);
+                }
+            } else {
+                records = all_records;
+            }
+            return records;
         }
     }
     else
     {
-        Console.WriteLine($"There are no cheeps to read from here: {path}\nPlease specify a different path, or simply begin cheeping to create some cheeps!");
-        return null;
+        throw new FileNotFoundException($"There are no cheeps to read from here: {path}\nPlease specify a different path, or simply begin cheeping to create some cheeps!");
```

Comment:
Do we still need the part of "specify new path"?

Reply:
It is for the sake of the developer, to indicate that the path is not found to the CSV file

Reply:
Makes sence, keep it.